google-cloud-bigquery/src/main/resources/META-INF/native-image/com.google.cloud/google-cloud-bigquery/reflect-config.json

```diff
@@ -3,8 +3,67 @@
     "name":"io.netty.buffer.AbstractByteBufAllocator",
     "queryAllDeclaredMethods":true
   },
+  {
+    "name":"io.netty.buffer.PooledByteBufAllocator",
```

Comment:
Do you intend to manually maintain this config file in the future? How much ongoing maintenance work is needed? Would we need to make these changes if we made code changes related to Arrow or add new Arrow dependencies? cc @prash-mi

Reply:
That's a good question. Given that we have a bit more background on GraalVM at the moment, we plan to maintain these configurations with client library owners being involved in the review process. However, we're currently working on making our onboarding documentation more robust as we learn more about the technology so that it is easier for client library owners to contribute and get unblocked sooner if needed in the future . Will keep you posted on the plans for that.  

> Would we need to make these changes if we made code changes related to Arrow or add new Arrow dependencies?

Most of these changes involved configurations for the netty classes that Apache Arrow brings in. These currently exist in `gax-grpc` but have been added here explicitly to avoid adding a dependency on gax-grpc. Since bigquery and most libraries bring in gax, we might want to consider moving them to `gax` or even closer to netty which could help make this file a little shorter. As far as the Arrow dependencies are concerned, I don't have a solid answer for this but there is a possibility that we'll need to add new configurations if new deps are added. Luckily, we have our Kokoro GraalVM presubmit checks to catch them if required.